src/node/rpc/member_frontend.h

```diff
@@ -330,6 +382,162 @@ namespace ccf
+        {
+          LOG_FAIL_FMT(
+            "Proposal {}: key id {} already added for different issuer",
+            proposal_id,
+            jwk.kid);
+          return false;
+        }
+        if (jwk.x5c.empty())
+        {
+          LOG_FAIL_FMT("Proposal {}: JWKS is invalid (empty x5c)", proposal_id);
+          return false;
+        }
+
+        auto& der_base64 = jwk.x5c[0];
+        auto der = tls::raw_from_b64(der_base64);
+
+        std::map<std::string, std::vector<uint8_t>> claims;
+        bool has_key_policy_sgx_claims =
+          issuer_metadata.key_policy.has_value() &&
+          issuer_metadata.key_policy.value().sgx_claims.has_value() &&
+          !issuer_metadata.key_policy.value().sgx_claims.value().empty();
+        if (
+          issuer_metadata.key_filter == JwtIssuerKeyFilter::SGX ||
+          has_key_policy_sgx_claims)
+        {
+          oe_verifier_initialize();
+          oe_verify_attestation_certificate_with_evidence(
+            der.data(),
+            der.size(),
+            oe_verify_attestation_certificate_with_evidence_cb,
+            &claims);
+        }
+
+        if (
+          issuer_metadata.key_filter == JwtIssuerKeyFilter::SGX &&
+          claims.empty())
+        {
+          LOG_INFO_FMT(
+            "Proposal {}: Skipping JWT signing key with kid {} (not OE "
+            "attested)",
+            proposal_id,
+            jwk.kid);
+          continue;
+        }
+
+        if (has_key_policy_sgx_claims)
+        {
+          for (auto& [claim_name, expected_claim_val_hex] :
+               issuer_metadata.key_policy.value().sgx_claims.value())
+          {
+            if (claims.find(claim_name) == claims.end())
+            {
+              LOG_FAIL_FMT(
+                "Proposal {}: JWKS kid {} is missing the {} SGX claim",
+                proposal_id,
+                jwk.kid,
+                claim_name);
+              return false;
+            }
+            auto& actual_claim_val = claims[claim_name];
+            auto actual_claim_val_hex =
+              fmt::format("{:02x}", fmt::join(actual_claim_val, ""));
+            if (expected_claim_val_hex != actual_claim_val_hex)
+            {
+              LOG_FAIL_FMT(
+                "Proposal {}: JWKS kid {} has a mismatching {} SGX claim",
+                proposal_id,
+                jwk.kid,
+                claim_name);
+              return false;
+            }
+          }
+        }
+        else
+        {
+          mbedtls_x509_crt cert;
+          mbedtls_x509_crt_init(&cert);
+          int rc = mbedtls_x509_crt_parse(&cert, der.data(), der.size());
+          if (rc != 0)
+          {
+            LOG_FAIL_FMT(
+              "Proposal {}: JWKS kid {} has an invalid X.509 certificate: "
+              "{}",
+              proposal_id,
+              jwk.kid,
+              tls::error_string(rc));
+            return false;
+          }
+          mbedtls_x509_crt_free(&cert);
```

Comment:
Minor: Could the `mbedtls`-specific code be moved to `tls/key_pair.h`. We also already parse certificates (see `public_key_pem_from_cert()`) so we can probably refactor this to only call into `mbedtls` once.